### PR TITLE
Initial Feat/timesheet refactor

### PR DESCRIPTION
**Pull request by raulfdm**
# Changes

## Removed

* deprecated useTimesheet

## Changed

* split admin and user functionality correctly between pages (`index.vue` / `_start_timestamp.vue`)

## How to test

* log in as user
* view/add/edit your timesheet

* log in as admin
* selected timesheets from burger menu
* view any timesheet / approve / reject timesheet

### CODE REVIEW

File: pages/index.vue [hunk 1]
```diff
@@ -6,220 +6,175 @@
...
     bridge: "Bridge"
     standByHours: "Stand-by uren"
     leaveDay: "Verlofdagen"
     addComment: "Notitie toevoegen"
-    denialReason: "Reden van afwijzing"
-    denialReasonPlaceholder: "Schrijf een korte notitie van de reden van afkeuren. Dit wordt getoond aan de medewerker."
 </i18n>
 <template>
   <div class="content-wrapper mt-5">
+    <h6>currentEmployee:</h6>
```
Comment (Cool-ka) [inline pages/index.vue:19]: we don't want to have it translated?

File: pages/index.vue [hunk 2]
```diff
@@ -6,220 +6,175 @@
...
+    <p>{{employee}}</p>
+
+    <h6>timesheet:</h6>
```
Comment (Cool-ka) [inline pages/index.vue:22]: this one as well

File: pages/index.vue [hunk 3]
```diff
@@ -263,146 +225,337 @@ export default defineComponent({
...
+      () => `${i18n.t('timesheets')} - ${currentEmployee.value?.name}`
+    );
+    const message = computed(
+      () => timesheetState.value?.timesheets[0]?.message
+    );
+    const showBridgeError = computed(() => {
+      return !!store.state.records.errorMessageWorkscheme;
+    });
+    const timesheetDenyMessage = computed(() =>
+      timesheetState.value.timesheets[0]
```
Comment (Cool-ka) [inline pages/index.vue:250]: maybe it is a personal preference, but I like to write it shorter
`timesheetState.value.timesheets[0] || ''`

File: pages/index.vue [hunk 4]
```diff
@@ -263,146 +225,337 @@ export default defineComponent({
...
-      timesheet.autoSave();
+      const newTimesheet = {
+        projects: previousWeekTimesheet.projects
+          .filter((project) => {
+            const projectArchived = customers.value.customers.find(
+              (customer) => customer.id === project.customer.id
+            )?.archived;
+            if (projectArchived) {
+              alert(
+                `${project.customer.name} is already archived. We wont copy it to new timesheet`
```
Comment (Cool-ka) [inline pages/index.vue:405]: is the proper grammar in those text important? Not sure if we should write 'won't' and 'to A new timesheet'

File: pages/index.vue [hunk 5]
```diff
@@ -263,146 +225,337 @@ export default defineComponent({
...
+      });
+
+      let reasonOfDenial = '';
+      if (timesheetState.value.timesheets[0]) {
+        reasonOfDenial = timesheetState.value.timesheets[0].reasonOfDenial;
+      }
+      if (newTimesheetStatus === recordStatus.DENIED && denialMessage) {
+        reasonOfDenial = denialMessage;
+      } else if (newTimesheetStatus === recordStatus.PENDING) {
+        reasonOfDenial = '';
```
Comment (Cool-ka) [inline pages/index.vue:497]: there is no chance that we would overwrite the reasonOfDenial here?

File: pages/index.vue [hunk 6]
```diff
@@ -6,220 +6,175 @@
...
-    <employee-header
-      v-if="isAdminView && employee"
-      class="mb-5"
-      :employee="employee"
-    />
-
     <navigation-buttons
       class="mb-5"
       :selected-week="recordsState.selectedWeek"
-      :is-admin-view="isAdminView"
```
Comment (Cool-ka) [inline pages/index.vue:36]: did you remove this from the navigation-buttons component as well?

File: pages/index.vue (hunk 5 above)
Comment (DeanMW) [reply pages/index.vue:497]: Good point Ive removed this since it does nothing.

**Pull request merged**